app/main/posts/views/mode-context-form-filter.directive.js

```diff
@@ -10,23 +10,29 @@ function ModeContextFormFilterDirective() {
     };
 }
 
-ModeContextFormFilter.$inject = ['$scope', 'FormEndpoint', 'PostEndpoint', '$q', '_', '$rootScope', 'PostSurveyService', '$timeout'];
-function ModeContextFormFilter($scope, FormEndpoint, PostEndpoint, $q, _, $rootScope, PostSurveyService, $timeout) {
+ModeContextFormFilter.$inject = ['$scope', 'FormEndpoint', 'PostEndpoint', '$q', '_', '$rootScope', 'PostSurveyService', 'PostFilters', '$timeout', '$location'];
+function ModeContextFormFilter($scope, FormEndpoint, PostEndpoint, $q, _, $rootScope, PostSurveyService, PostFilters, $timeout, $location) {
     $scope.forms = [];
     $scope.showOnly = showOnly;
     $scope.hide = hide;
     $scope.unknown_post_count = 0;
     $scope.hasManageSettingsPermission = $rootScope.hasManageSettingsPermission;
     $scope.canAddToSurvey = PostSurveyService.canCreatePostInSurvey;
-
+    $scope.PostFilters = PostFilters;
+    $scope.unmapped = 0;
+    $scope.location = $location;
+    $scope.goToUnmapped = goToUnmapped;
+    $scope.getUnmapped = getUnmapped;
     activate();
 
     function activate() {
         // Load forms
         $scope.forms = FormEndpoint.query();
         var postCountRequest = PostEndpoint.stats({ group_by: 'form', status: 'all' });
-
-        $q.all([$scope.forms.$promise, postCountRequest.$promise]).then(function (responses) {
+        // todo: can I limit this to only returning 'unmapped'?
```

Comment:
I think we need to look at our requests overall, particularly, in light of the needs of mobile. You could add a separate request route to return just the counts. I'll take a look at this later today and try to come up with a better answer.